Easydict/Swift/Service/Volcano/VolcanoService.swift

```diff
@@ -0,0 +1,178 @@
+//
+//  VolcanoService.swift
+//  Easydict
+//
+//  Created by Jerry on 2024-08-11.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import Alamofire
+import Defaults
+import Foundation
+
+@objc(EZVolcanoService)
+public final class VolcanoService: QueryService {
+    // MARK: Public
+
+    override public func serviceType() -> ServiceType {
+        .volcano
+    }
+
+    override public func link() -> String? {
+        "https://translate.volcengine.com"
+    }
+
+    override public func name() -> String {
+        NSLocalizedString("volcano_translate", comment: "The name of Volcano Translate")
+    }
+
+    override public func supportLanguagesDictionary() -> MMOrderedDictionary<AnyObject, AnyObject> {
+        let orderedDict = MMOrderedDictionary<AnyObject, AnyObject>()
+        for (key, value) in VolcanoTranslateType.supportLanguagesDictionary {
+            orderedDict.setObject(value as NSString, forKey: key.rawValue as NSString)
+        }
+        return orderedDict
+    }
+
+    override public func ocr(_: EZQueryModel) async throws -> EZOCRResult {
+        logInfo("Volcano Translate currently does not support OCR")
+        throw QueryServiceError.notSupported
+    }
+
+    override public func needPrivateAPIKey() -> Bool {
+        true
+    }
+
+    override public func hasPrivateAPIKey() -> Bool {
```

Comment:
The `hasPrivateAPIKey()` method usually does not need to be overridden; it only needs to be when the service has a built-in key.

The `needPrivateAPIKey()` is used in conjunction with the `hasPrivateAPIKey()` method, and there is no need to override them here; otherwise, strange prompt messages will appear when both `accessKeyId` and `secretAccessKey` are empty.

This part of the code was not designed very well initially, which is a bit confusing. I will try to optimize it later.

<img width="442" alt="49675" src="https://github.com/user-attachments/assets/4b50c264-ea31-4473-b6b3-05905a94608a">